frontend/server/src/main/java/org/pytorch/serve/wlm/ModelManager.java

```diff
@@ -205,87 +204,139 @@ private ModelArchive createModelArchive(
         return archive;
     }
 
+    private void setupModelVenv(Model model)
+            throws IOException, InterruptedException, ModelException {
+        String venvPath = EnvironmentUtils.getPythonVenvPath(model);
+        List<String> commandParts = new ArrayList<>();
+        commandParts.add(EnvironmentUtils.getPythonRunTime(model));
+        commandParts.add("-m");
+        commandParts.add("venv");
```

Comment:
should you make this name customizable? Part of the appeal of this PR is different workers should be able to have different virtual environments

Reply:
Currently this PR creates a virtual environment on a per model basis at model load time. All workers for a given model use the same virtual environment. This replaces installing dependencies on a per model basis in a target directory and is backwards compatible with the existing behavior with no change to customer experience. Although the same name `venv` is used, they are located within the individual model directories, for ex: `/tmp/models/test-model/venv`. Would it be useful to extend this implementation to support separate `venv` per worker?

Reply:
No the isolation is fine as is I think